### PR TITLE
SphereRenderer

### PR DESCRIPTION
**Pull request by braunms**
- Added FlagStorage support for all render modes
- Consolidated redundant shader code
- Added Ambient Occlusion to regression test
- Some code formatting and cleanup
- Table2Particles: Considering radius for bbox size calculation
- Transformed color param in ClippingPlane to 'real' ColorParam

### CODE REVIEW

**Comment by reinago** [comment]
can we consolidate this even more? I see many blah_attributes and they are pretty similar. also sometimes the blah start still contains attributes, doesn't it?

File: core/Shaders/sphere/splat_fragment_mainstart.glsl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+
+void main(void) {
```
Comment (tobiasrau) [inline core/Shaders/sphere/splat_fragment_mainstart.glsl]: Could this also become a simple_mainstart?

**Review by tobiasrau — CHANGES_REQUESTED** [review]

File: core/Shaders/sphere/splat_fragment_mainstart.glsl (hunk 1 above)
Comment (braunms) [reply core/Shaders/sphere/splat_fragment_mainstart.glsl]: Jupp, good point, done in commit a31ee39.

**Review by tobiasrau — APPROVED** [review]

**Review by tobiasrau — CHANGES_REQUESTED** [review]
1|[SphereRenderer] Unable to compile sphere shader (@compile vertex): 0:16(2): error: vertex shader input / attribute cannot have array type in GLSL 1.30 (GLSL 1.50 required)

**Pull request merged**